src/main/java/io/zeebe/kafka/connect/source/ZeebeSourceInflightRegistry.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2019 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.kafka.connect.source;
+
+import io.zeebe.client.api.response.ActivatedJob;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+class ZeebeSourceInflightRegistry {
+
+  private final Integer maxJobsToActivate;
+  private final List<String> jobTypes;
+  private final Map<String, Integer> inflightJobPerTypeCounts;
+  private final Deque<JobInfo> inflightJobInfoQueue;
```

Comment:
Out of curiosity, why a `Deque` and not a `Queue`? Isn't the behaviour here still FIFO?

Reply:
No specific reason, was just the remains of a previous approach, I will change it to Queue again